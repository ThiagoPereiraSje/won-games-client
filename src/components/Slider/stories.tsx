import { ComponentMeta, ComponentStory } from '@storybook/react'
import styled from 'styled-components'
import Slider from '.'

export default {
  title: 'Slider',
  component: Slider,
  args: {
    settings: {
      dots: true,
      infinite: true,
      speed: 500,
      slidesToShow: 1,
      slidesToScroll: 1
    }
  }
} as ComponentMeta<typeof Slider>

const Slide = styled.div`
  background-color: gray;
  width: 30rem;
  padding: 10rem 0;
  border: 0.1rem solid red;
  color: white;
  text-align: center;
`

export const Horizontal: ComponentStory<typeof Slider> = (args) => (
  <Slider {...args}>
    <Slide>1</Slide>
    <Slide>2</Slide>
    <Slide>3</Slide>
    <Slide>4</Slide>
    <Slide>5</Slide>
  </Slider>
)

export const Vertical: ComponentStory<typeof Slider> = (args) => (
  <Slider {...args}>
    <Slide>1</Slide>
    <Slide>2</Slide>
    <Slide>3</Slide>
    <Slide>4</Slide>
    <Slide>5</Slide>
  </Slider>
)

Vertical.args = {
  settings: {
    vertical: true,
    verticalSwiping: true,
    infinite: false,
    dots: true
  }
}
