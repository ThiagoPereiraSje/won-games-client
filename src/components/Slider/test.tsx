import { render, screen } from '@testing-library/react'
import Slider from '.'

describe('<Slider />', () => {
  it('should render the heading', () => {
    render(<Slider />)

    const element = screen.getByRole('heading', { name: /Slider/i })

    expect(element).toBeInTheDocument()
  })

  it('should render correctly', () => {
    const { container } = render(<Slider />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
