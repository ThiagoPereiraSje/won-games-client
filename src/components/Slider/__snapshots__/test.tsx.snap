// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slider /> should render correctly 1`] = `
<main
  class=""
>
  <h1>
    Hello Slider
  </h1>
</main>
`;
